client/src/components/TaskCard/TaskCard.js

```diff
@@ -35,6 +35,7 @@ const TaskCard = ({
 
   const reflectionDoneHandler = async (reflectionComments, satisfaction) => {
     await createTaskReflection(id, reflectionComments, satisfaction);
```

Comment:
Minor comment: I think it makes more sense to toggle the task first then create the task reflection rather than the other way around. (i.e. The task should be ended before you create a task reflection)

Reply:
I see, that makes sense.